app/Http/Controllers/Device/Tabs/PortsController.php

```diff
@@ -175,24 +175,22 @@ public function findPortNeighbors(Port $port): array
             }
         }
 
-        if ($this->detail) {
-            // IPv4 + IPv6 subnet if detailed
-            // fa-arrow-right green portlink on devicelink
-            if ($port->ipv4Networks->isNotEmpty()) {
-                $ids = $port->ipv4Networks->map(fn ($net) => $net->ipv4->pluck('port_id'))->flatten();
-                foreach ($ids as $port_id) {
-                    if ($port_id !== $port->port_id) {
-                        $this->addPortNeighbor($neighbors, 'ipv4_network', $port_id);
-                    }
+        // IPv4 + IPv6 subnet if detailed
+        // fa-arrow-right green portlink on devicelink
+        if ($port->macLinkedPorts->isNotEmpty()) {
```

Comment:
You need to change the eager loaded relationships drop ipv4Networks (if it is no longer used) and add macLinkedPorts.